Sources/FioriAR/ARCards/Views/ContentViews/ARAnnotationsView.swift

```diff
@@ -96,7 +112,10 @@ public struct ARAnnotationsView<Scan: View, Card: View, Marker: View, CardItem>:
             if arModel.discoveryFlowHasFinished {
                 ARAnnotationContentView(arModel, cardLabel: cardLabel, markerLabel: markerLabel)
             } else {
-                scanLabel(guideImage == nil ? arModel.guideImage : .finished(guideImage!), arModel.anchorPosition)
+                // Creative and perhaps confusing albeit it works
+                if let erasedGuideItem = ((guideImage as? GuideItem) ?? (arModel.guideImageType as? GuideItem)) {
```

Comment:
Developer might use `ARAnnotationsView` initializer with `scanLabel` ViewBuilder expecting `UIImage` while using asynchronous loading strategy (returning `GuideImageState`) or using `scanLabel` ViewBuilder expecting `GuideImageState` enum while using synchronous loading strategy (returning `UIImage`).

In both cases the `ARAnnotationsView` body implementation would not call `scanLabel` ViewBuilder due to type cast errors.

This can be fixed but would require 4 cases to be handled which is messy